relativity.scad

```diff
@@ -383,97 +383,101 @@ function _match_parsed_rx(string, regex, string_pos=0, ignore_case=false) =
 			string_pos+1
 		:
 			undef
-		
+
 	//LITERAL
 	: equals(string[string_pos], regex, ignore_case=ignore_case) ?
 		string_pos+1
-	
+
 	//WILDCARD
 	: regex == "."?
 		string_pos+1
-	
+
 	//NO MATCH
-	: 
+	:
 		undef
 	;
 
 
-function is_empty(string) = 
+function is_empty(string) =
 	string == "";
 
-function is_null_or_empty(string) = 
+function is_null_or_empty(string) =
 	string == undef || string == "";
-	
-function is_null_or_whitespace(string) = 
+
+function is_null_or_whitespace(string) =
 	string == undef || trim(string) == "";
 
-function trim(string) = 
+function trim(string) =
 	string == undef?
 		undef
 	: string == ""?
 		""
 	:
 		_null_coalesce(
-			between(string, _match_set(string, _whitespace, 0), 
+			between(string, _match_set(string, _whitespace, 0),
 					_match_set_reverse(string, _whitespace, len(string))),
 			""
 		)
 	;
 
-function _match_repetition(string, regex, min_reps, max_reps, pos, ignore_case=false) = 
+function opt_decr(opt_num) =
+  is_undef(opt_num) ? undef
+  : opt_num - 1 ;
+
+function _match_repetition(string, regex, min_reps, max_reps, pos, ignore_case=false) =
     pos == undef?
         undef
     : pos > len(string)?
         undef
 	: _null_coalesce(
-		_match_repetition(string, regex, min_reps-1, max_reps-1, 
-			_match_parsed_rx(string, regex, pos, ignore_case=ignore_case), 
+		_match_repetition(string, regex, opt_decr(min_reps), opt_decr(max_reps),
```

Comment:
this is the code change that was done to handle https://github.com/davidson16807/relativity.scad/issues/21